lcm-logger/lcm-logger.1

```diff
@@ -1,82 +1,114 @@
-Shows some help text and exits
+\fB\-h\fR, \fB\-\-help\fR
+Shows this help text and exits.
 .TP
-.B \-i, \-\-increment
-Automatically append a suffix to \fIFILE\fR such that the resulting filename
-does not already exist.  This option precludes -f and --rotate.
+\fB\-i\fR, \fB\-\-increment\fR
+Automatically append a suffix to FILE
+such that the resulting filename does not
+already exist.  This option precludes \fB\-f\fR and
+\fB\-\-rotate\fR.
 .TP
-.B \-l, \-\-lcm\-url=\fIURL\fR
+\fB\-l\fR, \fB\-\-lcm\-url\fR=\fI\,URL\/\fR
 Log messages on the specified LCM URL
 .TP
-.B \-m, \-\-max\-unwritten-mb=\fISIZE\fR
-Maximum size of received but unwritten messages to store in memory before
-dropping messages.  (default: 100 MB)
+\fB\-m\fR, \fB\-\-max\-unwritten\-mb\fR=\fI\,SZ\/\fR
+Maximum size of received but unwritten
+messages to store in memory before dropping
+messages.  (default: 100 MB)
 .TP
-.B \-\-rotate=\fINUM\fR
-When creating a new log file, rename existing files out of the way and always write to FILE.0.  If
-FILE.0 already exists, it is renamed to FILE.1.  If FILE.1 exists, it is
-renamed to FILE.2, etc.  If FILE.NUM exists, then it is deleted.  This option
-precludes -i.
+\fB\-\-rotate\fR=\fI\,NUM\/\fR
+When creating a new log file, rename existing files
+out of the way and always write to FILE.0.
+If FILE.0 already exists, it is renamed to FILE.1.
+If FILE.1 exists, it is renamed to FILE.2, etc.
+If FILE.NUM exists, then it is deleted.
+This option precludes \fB\-i\fR.
 .TP
-.B \-\-split\-mb=\fIN\fR
-Automatically start writing to a new log file once the log file exceeds N MB in size
-(can be fractional).  This option requires -i or --rotate.
+\fB\-\-split\-mb\fR=\fI\,N\/\fR
+Automatically start writing to a new log
+file once the log file exceeds N MB in size
+(can be fractional).  This option requires \fB\-i\fR
+or \fB\-\-rotate\fR.
 .TP
-.B \-q, \-\-quiet
+\fB\-q\fR, \fB\-\-quiet\fR
 Suppress normal output and only report errors.
 .TP
-.B \-s, \-\-strftime
-Format \fIFILE\fR with strftime.
+\fB\-a\fR, \fB\-\-append\fR
+Append events to the given log file.
 .TP
-.B \-v, \-\-invert-channels
-Invert channels.  Log everything that \fICHAN\fR does not match.
-
-.SH ROTATING AND SPLITTING
+\fB\-s\fR, \fB\-\-strftime\fR
+Format FILE with strftime.
+.TP
+\fB\-v\fR, \fB\-\-invert\-channels\fR
+Invert channels.  Log everything that CHAN
+does not match.
+.TP
+\fB\-\-disk\-quota\fR=\fI\,SIZE\/\fR
+Minimum amount of free space to reserve on the disk
+being written to. lcm\-logger will exit when it sees
+the current free disk space has fallen below the
+quota minimum.
+For example, given `du \fB\-H\fR` reports 15 GB free, and
+that nothing else is changing the target disk,
+\fB\-\-disk\-quota\fR=\fI\,10GB\/\fR means that lcm\-logger will not
+write more than 5 GB before it exits.
+Units accepted: B
+1024: K,   M,   G,   T,   P,   E,
+.TP
+Ki,
+Mi,  Gi,  Ti,  Pi,  Ei,
+.TP
+KiB, MiB, GiB, TiB, PiB, EiB
+1000: KB, MB, GB, TB, PB, EB
```

Comment:
Yes I get the same thing. It appears this is because the help message has some extra indentation which is messing up the help2man output. Unfortunately, help2man doesn't have a lot of options for customizing this sort of thing, at least as far as I can tell.

So then I think the options are:

- Just live with it; we'd have a reduced-quality man page but enjoy automatic generation and a nicely-formatted `--help` output
- Try to postprocess it. It would significantly complicate the generation process, but I could try to `sed` the output to catch and improve this case.
- Make this section hard-coded. It could be added to the input file and hard-coded however we like

Any preference?

Reply:
>Just live with it; we'd have a reduced-quality man page but enjoy automatic generation and a nicely-formatted --help output

Of those options, I think this is my preference.  Although, I wonder if modifying the help output from `lcm-logger` is an option.  It looks like that's the source of the issue.  What do you think about this change to `lcm-logger.c`,

```diff
diff --git a/lcm-logger/lcm_logger.c b/lcm-logger/lcm_logger.c
index 4d216fb..09e49e8 100644
--- a/lcm-logger/lcm_logger.c
+++ b/lcm-logger/lcm_logger.c
@@ -618,9 +618,8 @@ static void usage()
             "                             --disk-quota=10GB means that lcm-logger will not\n"
             "                             write more than 5 GB before it exits.\n"
             "                             Units accepted: B\n"
-            "                             1024: K,   M,   G,   T,   P,   E,\n"
-            "                                   Ki,  Mi,  Gi,  Ti,  Pi,  Ei,\n"
-            "                                   KiB, MiB, GiB, TiB, PiB, EiB\n"
+            "                             1024: K, M, G, T, P, E, Ki, Mi, Gi, Ti, Pi, Ei, KiB, "
+            "MiB, GiB, TiB, PiB, EiB\n"
             "                             1000: KB, MB, GB, TB, PB, EB\n"
             "                             e.g. \"50G\" or \"0.05 TiB\". \n"
             "                             Units are treated case insensitively. gb is treated\n"
```

It produces this `--help` output,

```
      --disk-quota=SIZE      Minimum amount of free space to reserve on the disk
                             being written to. lcm-logger will exit when it sees
                             the current free disk space has fallen below the
                             quota minimum.
                             For example, given `du -H` reports 15 GB free, and
                             that nothing else is changing the target disk,
                             --disk-quota=10GB means that lcm-logger will not
                             write more than 5 GB before it exits.
                             Units accepted: B
                             1024: K, M, G, T, P, E, Ki, Mi, Gi, Ti, Pi, Ei, KiB, MiB, GiB, TiB, PiB, EiB
                             1000: KB, MB, GB, TB, PB, EB
                             e.g. "50G" or "0.05 TiB".
                             Units are treated case insensitively. gb is treated
                             as GB (gigabytes) and not as gigabits.
                             Note for `df`: -h is 1024 and -H is 1000 units.
                             NOT currently supported on Windows!
```

and this `man` page,

```
       --disk-quota=SIZE
              Minimum amount of free space to reserve on the disk being written to. lcm-logger will
              exit when it sees the current free disk space has fallen  below  the  quota  minimum.
              For  example, given `du -H` reports 15 GB free, and that nothing else is changing the
              target disk, --disk-quota=10GB means that lcm-logger will not write more  than  5  GB
              before  it  exits.  Units accepted: B 1024: K, M, G, T, P, E, Ki, Mi, Gi, Ti, Pi, Ei,
              KiB, MiB, GiB, TiB, PiB, EiB 1000: KB, MB, GB, TB, PB, EB e.g. "50G" or  "0.05  TiB".
              Units  are treated case insensitively. gb is treated as GB (gigabytes) and not as gi‐
              gabits.  Note for `df`: -h is 1024 and -H is 1000 units.  NOT currently supported  on
              Windows!
```

Reply:
I think it significantly improves the formatting in the man page and slightly worsens the formatting in the help message. Overall I think it's a good compromise. Implemented in 4bb4358dfd0a8ff1679daa019f8559f744dbf23e